primitives/io/src/lib.rs

```diff
@@ -142,6 +142,22 @@ pub trait Storage {
 	fn next_key(&mut self, key: &[u8]) -> Option<Vec<u8>> {
 		self.next_storage_key(&key)
 	}
+
+	/// Push new entry to the storage accumulator.
+	fn accumulator_push(&mut self, key: &[u8], appended: &[u8]) {
```

Comment:
I generally find these slice vs. vec arguments debatable and uncertain. AFAIK more types can be represented as slice which is good but if the underlying function wants a vec, then all those slices can also be a vec in the first place?

Reply:
Yeah, here `appended` should be a `Vec`. Will decrease the number of clones.